src/main/java/edu/ucsb/cs156/rec/controllers/RecommendationRequestController.java

```diff
@@ -0,0 +1,162 @@
+     */
+    @Operation(summary = "Get a single recommendation request")
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @GetMapping("")
+    public RecommendationRequest getById(
+            @Parameter(name = "id") @RequestParam Long id) {
+        RecommendationRequest recommendationRequest = recommendationRequestRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException(RecommendationRequest.class, id));
+
+        return recommendationRequest;
+    }
+
+    /**
+     * Create a new recommendation request
+     * 
+     * @param professorName       the name of the professor
+     * @param professorEmail      the email of the professor
+     * @param requesterName       the name of the requester
+     * @param recommendationTypes the types of recommendations
+     * @param details             the details of the request
+     * @param status              the status of the request
+     * @param submissionDate      the submission date of the request
+     * @param completionDate      the completion date of the request
+     * @return the saved recommendation request
+     */
+
+    @Operation(summary = "Create a new recommendation request")
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @PostMapping("/post")
+    public RecommendationRequest postRecommendationRequest(
+            @Parameter(name = "professorName") @RequestParam String professorName,
+            @Parameter(name = "professorEmail") @RequestParam String professorEmail,
+            @Parameter(name = "requesterName") @RequestParam String requesterName,
+            @Parameter(name = "recommendationTypes") @RequestParam String recommendationTypes,
+            @Parameter(name = "details") @RequestParam String details,
+            @Parameter(name = "status") @RequestParam String status,
+            @Parameter(name = "submissionDate", description = "date (in iso format, e.g. YYYY-mm-ddTHH:MM:SS; see https://en.wikipedia.org/wiki/ISO_8601)") @RequestParam("submissionDate") @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) LocalDateTime submissionDate,
+            @Parameter(name = "completionDate", description = "date (in iso format, e.g. YYYY-mm-ddTHH:MM:SS; see https://en.wikipedia.org/wiki/ISO_8601)") @RequestParam(value = "completionDate", required = false) @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) LocalDateTime completionDate)
+            throws JsonProcessingException {
+        // For an explanation of @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME)
+        // See: https://www.baeldung.com/spring-date-parameters
+
+        log.info("localDateTime={}", submissionDate);
+        log.info("localDateTime={}", completionDate);
+
+        RecommendationRequest recommendationRequest = new RecommendationRequest();
+        recommendationRequest.setProfessorName(professorName);
+        recommendationRequest.setProfessorEmail(professorEmail);
+        recommendationRequest.setRequesterName(requesterName);
```

Comment:
```suggestion
        recommendationRequest.setRequesterName(currentUser.getName());
```

And where is the requesterEmail?  Requester name can be ambiguous.  Do you know how many "David Wang" students there are that I've taught over the years, sometimes at the same time?

Whether you use userId from the user's table (a better approach, in my view) or email, it's essential to use some way of identifying users that is guaranteed to be unique.   Name is not.